perses/perses
# Runtime Plugins

# Description

This PR is a PoC for the new runtime plugins architecture. The core plugins involved in e2e tests were extracted as a runtime plugin living in: https://github.com/perses/perses-plugins

# Screenshots



# Checklist

  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.

## UI Changes

Use a Dynamic panel to load plugins at runtime

## How to test
- run perses ui
```bash
cd ui && npm run start
```
- run perses backend for development, this will install the default plugins from https://github.com/perses/perses-plugins
```bash
./scripts/api_backend_dev.sh
```
- check Perses UI. **Note** that only the following plugins will be available:
  - BarChart
  - GaugeChart
  - Markdown
  - StatChart
  - TimeSeriesChart
